src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,231 @@
+    */
+  def execIfAvailable(script: Path, args: String*): Unit = execIfAvailable(script.toString, args: _*)
+
+  /** Executes a script from the classpath, raise an exception otherwise.
+    *
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @throws ToolException when the exit code from the called process is not zero.
+    * @param scriptResource the name of the script resource on the classpath
+    * @param args a variable list of arguments to pass to the script
+    */
+  def exec(scriptResource: String, args: String*): Unit = exec(writeResourceToTempFile(scriptResource), args: _*)
+
+  /** Executes a script from the filesystem path.
+    *
+    * @throws Exception when we are unable to execute the script with the given arguments
+    * @throws ToolException when the exit code from the called process is not zero
+    * @param script Path to the script to be executed
+    * @param args a variable list of arguments to pass to the script
+    */
+  def exec(script: Path, args: String*): Unit =  {
+    val basename = PathUtil.basename(script, trimExt = false)
+    logger.info(s"Executing $basename with $executable using the arguments: ${args.mkString(" ")}")
+
+    val command  = executable +: script.toAbsolutePath.toString +: args
+    val process  = new ProcessBuilder(command:_*).redirectErrorStream(false).start()
+    val pipe1    = Io.pipeStream(process.getErrorStream, logger.warning)
+    val pipe2    = Io.pipeStream(process.getInputStream, logger.debug)
+    val exitCode = process.waitFor()
+    pipe1.close()
+    pipe2.close()
+    if (exitCode != 0) throw ToolException(exitCode)
+  }
+
+  /** Extracts a resource from the classpath and writes it to a temp file on disk.
+    *
+    * @param resource a given name on the classpath
+    * @return path to the temporary file
+    * */
+  private def writeResourceToTempFile(resource: String): Path = {
+    val lines = Io.readLinesFromResource(resource)
+    val dir   = Io.makeTempDir(PathUtil.sanitizeFileName(this.getClass.getSimpleName))
+    val path  = PathUtil.pathTo(dir.toString, PathUtil.basename(resource, trimExt = false))
+    dir.toFile.deleteOnExit()
+    Io.writeLines(path, lines)
+    path
+  }
+}
+
+/** Defines values used to get the version of the executable. */
+trait Versioned {
+  self: CommandLineTool =>
+
+  /** The default version flag. */
+  val versionFlag: String   = "--version"
+
+  /** Argument used to test version of the executable. */
+  val testArgs: Seq[String] = Seq(versionFlag)
+
+  /** Version of this executable. */
+  lazy val version: Try[ListBuffer[String]] = execCommand(executable +: testArgs: _*)
+}
+
+/** Defines methods used to check if specific modules are installed with the executable . */
+trait Modular {
+  self: CommandLineTool =>
+
+  /** Command used to test if a module is included with the tested executable. */
+  def TestModuleCommand(module: String): Seq[String]
```

Comment:
```suggestion
  def testModuleCommand(module: String): Seq[String]
```
https://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods

Likewise, lowercase:

- `ModuleAvailable`